tests/e2e/framework/istioctl.go

```diff
@@ -118,6 +119,27 @@ func (i *Istioctl) run(args string) error {
 	return nil
 }
 
+// RunWithOutput runs istioctl with the named args, optionally overriding the namespace with `ns` and the
+// istio namespace with `istioNs`.  `expectFailure` suppresses logging of failure when the command is
+// expected/desired to fail by the test.
+func (i *Istioctl) RunWithOutput(args []string, ns string, istioNs string, expectFailure bool) (output string, err error) {
+	if ns == "" {
+		ns = i.namespace
+	}
+	if istioNs == "" {
+		istioNs = i.istioNamespace
+	}
+
+	if output, err = util.Shell(fmt.Sprintf("%s --namespace %s --istioNamespace %s %s",
+		i.binaryPath, ns, istioNs, strings.Join(args, " "))); err != nil {
+		if !expectFailure {
```

Comment:
`RunWithOutput` is fine. I was commenting on the `expectFailure` flag to log an error vs. having the test handle it.